deployment/helm/templates/custom-configuration.yaml

```diff
@@ -6,7 +6,19 @@ metadata:
 spec:
   discoveryHandler: 
     name: {{ required "A custom.configuration.discoveryHandlerName is required." .Values.custom.configuration.discoveryHandlerName }}
-    discoveryDetails: {{ .Values.custom.configuration.discoveryDetails }}
```

Comment:
for custom-configuration,  i think the intent is that .Values.custom.configuration.discoveryDetails be yaml that can implicitly be what you are describing explicitly below.  if we make the explicit changes you have here, custom-configuration becomes coap-configuration (and can no longer be other configs)

Reply:
this seems more like the start of a deployment/helm/templates/coap-configuration.yaml :)

Reply:
Also the `custom.configuration` values should be added to a `coap` section of the values file: https://github.com/deislabs/akri/blob/main/deployment/helm/values.yaml. A `coap-discovery-handler.yaml` file should also be added.